test/core/event_engine/iomgr_event_engine/timer_list_test.cc

```diff
@@ -0,0 +1,246 @@
+            CheckResult::kTimersFired);
+  for (int i = 10; i < 20; i++) {
+    Mock::VerifyAndClearExpectations(&closures[i]);
+  }
+
+  EXPECT_CALL(host, Now())
+      .WillOnce(Return(kStart + grpc_core::Duration::Milliseconds(1600)));
+  EXPECT_EQ(FinishCheck(timer_list.TimerCheck(nullptr)),
+            CheckResult::kCheckedAndEmpty);
+}
+
+/* Cleaning up a list with pending timers. */
+TEST(TimerListTest, Destruction) {
+  Timer timers[5];
+  StrictMock<MockClosure> closures[5];
+
+  StrictMock<MockHost> host;
+  EXPECT_CALL(host, Now())
+      .WillOnce(
+          Return(grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(0)));
+  TimerList timer_list(&host);
+
+  EXPECT_CALL(host, Now())
+      .WillOnce(
+          Return(grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(0)));
+  timer_list.TimerInit(
+      &timers[0], grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(100),
+      &closures[0]);
+  EXPECT_CALL(host, Now())
+      .WillOnce(
+          Return(grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(0)));
+  timer_list.TimerInit(
+      &timers[1], grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(3),
+      &closures[1]);
+  EXPECT_CALL(host, Now())
+      .WillOnce(
+          Return(grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(0)));
+  timer_list.TimerInit(
+      &timers[2], grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(100),
+      &closures[2]);
+  EXPECT_CALL(host, Now())
+      .WillOnce(
+          Return(grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(0)));
+  timer_list.TimerInit(
+      &timers[3], grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(3),
+      &closures[3]);
+  EXPECT_CALL(host, Now())
+      .WillOnce(
+          Return(grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(0)));
+  timer_list.TimerInit(
+      &timers[4], grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(1),
+      &closures[4]);
+  EXPECT_CALL(host, Now())
+      .WillOnce(
+          Return(grpc_core::Timestamp::FromMillisecondsAfterProcessEpoch(2)));
+  EXPECT_CALL(closures[4], Run());
+  EXPECT_EQ(FinishCheck(timer_list.TimerCheck(nullptr)),
+            CheckResult::kTimersFired);
+  Mock::VerifyAndClearExpectations(&closures[4]);
+  EXPECT_FALSE(timer_list.TimerCancel(&timers[4]));
+  EXPECT_TRUE(timer_list.TimerCancel(&timers[0]));
+  EXPECT_TRUE(timer_list.TimerCancel(&timers[3]));
+  EXPECT_TRUE(timer_list.TimerCancel(&timers[1]));
+  EXPECT_TRUE(timer_list.TimerCancel(&timers[2]));
+}
+
+/* Cleans up a list with pending timers that simulate long-running-services.
+   This test does the following:
+    1) Simulates grpc server start time to 25 days in the past (completed in
+        `main` using TestOnlyGlobalInit())
+    2) Creates 4 timers - one with a deadline 25 days in the future, one just
+        3 milliseconds in future, one way out in the future, and one using the
+        grpc_timespec_to_millis_round_up function to compute a deadline of 25
```

Comment:
I think this should refer to `grpc_core::Timestamp::FromTimespecRoundUp()` instead.

Reply:
done